### ISSUE TITLE
sms notification not showing on desktop

### ISSUE THREAD

**Issue by philipraets** [andyholmes/gnome-shell-extension-gsconnect#318]
Hi,

I really like this program, but I have a problem with incoming sms.

I've got the latest version installed on openSUSE Tumbleweed with gnome 3.30.1, but my sms notifications aren't getting through for a while now. I can't find what's wrong. Sms-plugin is activated on the desktop and my phone.

this is what I have in the debugger:
```
DEBUG: [daemon.js:vfunc_startup:746]: Loading 1 device(s) from cache
DEBUG: [device.js:loadPlugin:768]: Samsung Galaxy Note 4: loading 'battery' plugin
DEBUG: [device.js:loadPlugin:768]: Samsung Galaxy Note 4: loading 'clipboard' plugin
DEBUG: [device.js:loadPlugin:768]: Samsung Galaxy Note 4: loading 'contacts' plugin
DEBUG: [device.js:loadPlugin:768]: Samsung Galaxy Note 4: loading 'findmyphone' plugin
DEBUG: [device.js:loadPlugin:768]: Samsung Galaxy Note 4: loading 'mousepad' plugin
DEBUG: [device.js:loadPlugin:768]: Samsung Galaxy Note 4: loading 'mpris' plugin
DEBUG: [device.js:loadPlugin:768]: Samsung Galaxy Note 4: loading 'notification' plugin
DEBUG: [device.js:loadPlugin:768]: Samsung Galaxy Note 4: loading 'ping' plugin
DEBUG: [device.js:loadPlugin:768]: Samsung Galaxy Note 4: loading 'runcommand' plugin
DEBUG: [device.js:loadPlugin:768]: Samsung Galaxy Note 4: loading 'sftp' plugin
DEBUG: [device.js:loadPlugin:768]: Samsung Galaxy Note 4: loading 'share' plugin
DEBUG: [device.js:loadPlugin:768]: Samsung Galaxy Note 4: loading 'sms' plugin
DEBUG: [device.js:loadPlugin:768]: Samsung Galaxy Note 4: loading 'systemvolume' plugin
DEBUG: [device.js:loadPlugin:768]: Samsung Galaxy Note 4: loading 'telephony' plugin
g_variant_new_string: assertion 'string != NULL' failed
DEBUG: [lan.js:_onIncomingChannel:98]: remote address: 192.168.51.11
DEBUG: [device.js:_setConnected:302]: Connected to Samsung Galaxy Note 4 (3309cc44ed24ccef)

DEBUG: [core.js:send:450]: Samsung Galaxy Note 4: {
  "id": 1541764757198,
  "type": "kdeconnect.sms.request_conversations",
  "body": {}
}

DEBUG: [core.js:receive/<:404]: Samsung Galaxy Note 4: {
  "id": 1541765146299,
  "type": "kdeconnect.telephony",
  "body": {
    "event": "sms",
    "messageBody": "Hello",
    "contactName": "My contact",
    "phoneThumbnail": "",
    "phoneNumber": "phoneNumber"
  }
}

DEBUG: [lan.js:_onIncomingIdentity:198]: {
  "id": 1541765148181,
  "type": "kdeconnect.identity",
  "body": {
    "deviceId": "3309cc44ed24ccef",
    "deviceName": "Samsung Galaxy Note 4",
    "protocolVersion": 7,
    "deviceType": "phone",
    "incomingCapabilities": [
      "kdeconnect.telephony.request_mute",
      "kdeconnect.contacts.request_vcards_by_uid",
      "kdeconnect.notification.request",
      "kdeconnect.clipboard",
      "kdeconnect.sftp.request",
      "kdeconnect.telephony.request",
      "kdeconnect.notification.reply",
      "kdeconnect.systemvolume",
      "kdeconnect.notification",
      "kdeconnect.battery.request",
      "kdeconnect.ping",
      "kdeconnect.sms.request_conversation",
      "kdeconnect.share.request",
      "kdeconnect.mpris",
      "kdeconnect.sms.request_conversations",
      "kdeconnect.contacts.request_all_uids_timestamps",
      "kdeconnect.sms.request",
      "kdeconnect.mousepad.request",
      "kdeconnect.runcommand",
      "kdeconnect.findmyphone.request"
    ],
    "outgoingCapabilities": [
      "kdeconnect.mousepad.echo",
      "kdeconnect.sms.messages",
      "kdeconnect.runcommand.request",
      "kdeconnect.mpris.request",
      "kdeconnect.notification.request",
      "kdeconnect.clipboard",
      "kdeconnect.notification",
      "kdeconnect.ping",
      "kdeconnect.sftp",
      "kdeconnect.contacts.response_uids_timestamps",
      "kdeconnect.share.request",
      "kdeconnect.telephony",
      "kdeconnect.mousepad.request",
      "kdeconnect.contacts.response_vcards",
      "kdeconnect.mousepad.keyboardstate",
      "kdeconnect.systemvolume.request",
      "kdeconnect.findmyphone.request",
      "kdeconnect.battery"
    ],
    "tcpPort": 1716,
    "tcpHost": "192.168.51.11"
  }
}

```
Also in the debug log, the incoming sms comes up 11 times, everytime with a different id.

What can I do to find the problem?

Thx,

Philip

**Comment by andyholmes** [comment]
Hi, and thanks for the debug log.

**TL;DR** What you want to do is enable the notification plugin on Android (and not disable the notification plugin in GSConnect). Also the contacts & sms plugin on Android (and leave them on in GSConnect).

One thing you may want to do, is open the SMS window and use that lovely hidden app-menu in the panel and select "Sync Contacts" and then "Sync Messages". Give it a second, then close and re-open the window. Sometimes this doesn't work automatically, it's a known bug, and upstream is working on it.

**Long Story**

A bit of back story. In the past there was a SMS _notification_ packet (which is the notification you see in Android) and an SMS _event_ packet which is this:

```js
{
  "id": 1541765146299,
  "type": "kdeconnect.telephony",
  "body": {
    "event": "sms",
    "messageBody": "Hello",
    "contactName": "My contact",
    "phoneThumbnail": "",
    "phoneNumber": "phoneNumber"
  }
}
```
If you've been using this for awhile, you may remember that sometimes resulted in duplicate notifications. Additionally, the SMS events are only sent once, when they happen, so if you miss them you don't get them.

The SMS _notifications_ are persistent (until you close them on either side) and also carry a `requestReplyId` allowing you to "quick reply". They look something like this:

```js
{
  "id": 1541767166010,
  "type": "kdeconnect.notification",
  "body": {
    "payloadHash": "ec6586b6d0c86cafae7808258924432a",
    "requestReplyId": "f99ff93d-247e-4892-9d47-5895203b6532",
    "id": "0|com.google.android.apps.messaging|0|com.google.android.apps.messaging:sms:1|10079",
    "appName": "Messages",
    "isClearable": true,
    "ticker": "My contact: Hello",
    "title": "My contact",
    "text": "Hello",
    "time": "1541767165992"
  },
  "payloadSize": 8082,
  "payloadTransferInfo": {
    "port": 1750
  }
}
```

As of the latest Android version (v1.10) SMS has it's own plugin and can provide history, as well as new messages; they sometimes come as a "digest" of all messages and look like this:

```js
{
  "id": 1541767163679,
  "type": "kdeconnect.sms.messages",
  "body": {
    "messages": [
      {
        "address": "555-555-5555",
        "body": "Hello",
        "date": 1541767162018,
        "type": 1, // <--- 0 or 1 (incoming/outgoing)
        "read": 0,
        "thread_id": 3, // <--- "conversation"
        "_id": 277, // <--- message
        "event": "sms"
      }
    ],
    "event": "batch_messages"
  }
}
```

The SMS plugin is still *very* new, which is why KDE Connect has not yet released their desktop-side version of it yet (I'm pretty sure) and may have a few bugs, but it's much easier to work with, determine the proper contact, and so on. It's okay if you receive a ton of _these_ duplicates because GSConnect can figure that out.

There's some ongoing work in #288 you can follow if you want, where we are stripping out all the old flaky code and re-designing the UI. It might be awhile before that is released (or even merged), but git usually quite stable these days (instructions in the Wiki) and you can always file bugs against the `master` branch if you want.

**Comment by philipraets** [comment]
I have enabled all plugins (both on phone and in GSConnect), but sms is still not working, everything else works fine. I did a clean install of openSUSE in KVM, and paired with it, but it has the same result

Also, if I open an sms window on the desktop, I can send sms, but the reply's don't come through (also not syncing messages)

I have reinstalled KDEconnect on the phone, reinstalled GSconnect on the desktop (also deleted the cache and config folders), but I always get the same result

**Comment by andyholmes** [comment]
Are there any errors in GNOME Logs about this? If GSConnect is actually receiving these packets, and a window is open for the same number, there should be errors thrown.

If you check the file `~/.cache/gsconnect/<device-id>/sms.json` is it filled with your messages? The files in the cache directory are just hardcopies of what's in memory, so either the plugins or GSConnect must be stopped for deleting them to have any effect.

**Comment by andyholmes** [comment]
Okay, it seems like there is something going wrong with the app. Especially if you are not receiving packet's like these:

```js
{
  "id": 1541767163679,
  "type": "kdeconnect.sms.messages",
  "body": {
    "messages": [
      {
        "address": "555-555-5555",
        "body": "Hello",
        "date": 1541767162018,
        "type": 1, // <--- 0 or 1 (incoming/outgoing)
        "read": 0,
        "thread_id": 3, // <--- "conversation"
        "_id": 277, // <--- message
        "event": "sms"
      }
    ],
    "event": "batch_messages"
  }
}
```
These are the types of packets we use for SMS communication now (although we do strip information from those older ones). If you go into the _App Info -> Permissions_ then use the top-right menu to select *All Permissions*, you should see a page like this (this is Oreo, other versions probably look similar):

![screenshot_20181114-043447](https://user-images.githubusercontent.com/1265208/48483188-a191df80-e7c7-11e8-8f4f-4547f8797e10.png)

Can you confirm these permissions are all enabled?

**Comment by philipraets** [comment]
These are the settings on the phone (Samsung galaxy note 4, android 6.0.1):
![screenshot_2018-11-14-13-56-37](https://user-images.githubusercontent.com/708011/48484372-3cb1a600-e816-11e8-8305-f911cbb1e4c5.png)
![screenshot_2018-11-14-13-51-28](https://user-images.githubusercontent.com/708011/48484379-3fac9680-e816-11e8-915b-f039e1df5023.png)
![screenshot_2018-11-14-13-53-59](https://user-images.githubusercontent.com/708011/48484383-42a78700-e816-11e8-825c-2ee78a424211.png)

And on the computer:
![screenshot from 2018-11-14 14-06-44](https://user-images.githubusercontent.com/708011/48484529-9b771f80-e816-11e8-9793-6b8ea8e5dab8.png)

(If I want to open the about screen, I get the following:

```
JS ERROR: Error: Wrong type number; string expected
_init/Gtk.Widget.prototype._init@resource:///org/gnome/gjs/modules/overrides/Gtk.js:60:9
_aboutAction@/home/BOMA/praets/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/daemon.js:399:27
@/home/BOMA/praets/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/daemon.js:881:2
```
)

**Comment by andyholmes** [comment]
> If I want to open the about screen, I get the following

Ah yep, a number of users are reporting this. It is fixed in the [latest release v16](https://github.com/andyholmes/gnome-shell-extension-gsconnect/releases/tag/v16), although I'm unsure why or when it started happening.

It seems you have all the same plugins and permissions as me, so I'm unsure why you aren't receiving SMS history. Are you using the default AOSP or Google Messages apps for SMS? Another user has reported SMS history does not work with Signal, possibly because it does not use Android's database for storing messages.

**Comment by andyholmes** [comment]
Unsurprisingly, Samsung doesn't have a repo for their messaging app, but a painfully vague Google search seems to indicate they use their own database.

I'll have to start compiling a list of apps that don't support it. So far, I know that Google Messages, the AOSP app and something called Textra will work.

**Comment by andyholmes** [comment]
I'm going to mark this a duplicate of #320 where I'll start collecting a list of messaging apps that don't work. Sorry it took so long to figure out, but thanks for digging into the logs!

duplicate of #320

**Comment by JohannPopper** [comment]
I'm not convinced this needs to be closed because it's certainly not a duplicate of #320. This isn't a history issue. It's the complete lack of SMS notifications since v13 of GSConnect on a range of Samsung phones using the default texting app. I have now tested KDE Connect on the latest KDE 5.14 and there is no problem receiving SMS notifications on desktop, so if GSConnect is based on the latest KDE Connect since the v13+ rewrite, something is definitely wrong in GSConnect and it can't be an upstream issue.

**Comment by andyholmes** [comment]
Okay, can you please enable [Debugging](https://github.com/andyholmes/gnome-shell-extension-gsconnect/wiki/Debugging), be sure *Notifications*, *SMS* and *Telephony* plugins are enabled in both GSConnect and Android, then tell me which of the following packets print when you receive an SMS?

**1** *Notification*
```js
{
  "id": 1541767166010,
  "type": "kdeconnect.notification",
  "body": {
    "payloadHash": "ec6586b6d0c86cafae7808258924432a",
    "requestReplyId": "f99ff93d-247e-4892-9d47-5895203b6532",
    "id": "0|com.google.android.apps.messaging|0|com.google.android.apps.messaging:sms:1|10079",
    "appName": "Messages",
    "isClearable": true,
    "ticker": "My contact: Hello",
    "title": "My contact",
    "text": "Hello",
    "time": "1541767165992"
  },
  "payloadSize": 8082,
  "payloadTransferInfo": {
    "port": 1750
  }
}
```

**2** *SMS*
```js
{
  "id": 1541767163679,
  "type": "kdeconnect.sms.messages",
  "body": {
    "messages": [
      {
        "address": "555-555-5555",
        "body": "Hello",
        "date": 1541767162018,
        "type": 1,
        "read": 0,
        "thread_id": 3,
        "_id": 277,
        "event": "sms"
      }
    ],
    "event": "batch_messages"
  }
}
```

**3** *Telephony*
```js
{
  "id": 1541765146299,
  "type": "kdeconnect.telephony",
  "body": {
    "event": "sms",
    "messageBody": "Hello",
    "contactName": "My contact",
    "phoneThumbnail": "",
    "phoneNumber": "phoneNumber"
  }
}
```

**Comment by JohannPopper** [comment]
Certainly. A million thanks for taking another look at this. I'm on Ubuntu 18.10, by the way, latest GSConnect to test this. Also, I tried a few older Samsung phones, a Galaxy S7, S6, and a very old S3, both of which (with the latest KDE Connect app) push texts to my Kubuntu 18.10 desktop. 

In the GSConnect debug console, I see 1 and 3 only, not 2. Each received SMS showed, in order, the following pair in console: 

}
DEBUG: [core.js:receive/<:404]: Samsung SAMSUNG-SM-J120A: {
  "id": 1544520638633,
  "type": "kdeconnect.telephony",
  "body": {
    "event": "sms",
    "messageBody": "FRM:sms@mc.internetmailserver.net,+ADw-sms+A\nMSG:",
    "phoneNumber": "1410200505"
  }
}
DEBUG: [core.js:receive/<:404]: Samsung SAMSUNG-SM-J120A: {
  "id": 1544520639033,
  "type": "kdeconnect.notification",
  "body": {
    "id": "0|com.android.mms|123|null|10040",
    "isCancel": true
  }

No text notification appeared on desktop.

**Comment by andyholmes** [comment]
So this here is the older, deprecated SMS packet, which is sent by the *Telephony* plugin. This was also deprecated in the [KDE Connect desktop client on Sep 17](https://github.com/KDE/kdeconnect-kde/commit/64062b3cc9bea4b0dcd13a1a0cd3ade5dcbdad83). You can see [here](https://github.com/KDE/kdeconnect-kde/blob/master/plugins/telephony/telephonyplugin.cpp#L110-L112) where it will bail if the packet is an SMS event.

According to the [release tags](https://github.com/KDE/kdeconnect-kde/tags) this should be in 1.3.2+, but that might not be in your distro yet (if sshfs doesn't work that's a strong sign this is the case).

Previously, this was used to show desktop notifications, and I believe KDE Connect used the "duplicate" *Notification* plugin notification in their Plasma interface.

```js
DEBUG: [core.js:receive/<:404]: Samsung SAMSUNG-SM-J120A: {
"id": 1544520638633,
"type": "kdeconnect.telephony",
"body": {
"event": "sms",
"messageBody": "FRM:sms@mc.internetmailserver.net,+ADw-sms+A\nMSG:",
"phoneNumber": "1410200505"
}
```

This next one is a *Notification* packet indicating that this notification has been closed on the remote device (`"isCancel": true`).

```js
DEBUG: [core.js:receive/<:404]: Samsung SAMSUNG-SM-J120A: {
"id": 1544520639033,
"type": "kdeconnect.notification",
"body": {
"id": "0|com.android.mms|123|null|10040",
"isCancel": true
}
```

Did you have the phone on with the messaging app open when you received this? If so it may have pre-empted the notification packet from being sent. If not, there should have been a notification packet, possibly before the above *Telephony* packet. It would have looked something like this:

```js
DEBUG: [core.js:receive/<:404]: Samsung SAMSUNG-SM-J120A: {
"id": 1544520639033,
"type": "kdeconnect.notification",
"body": {
    "payloadHash": "ec6586b6d0c86cafae7808258924432a",
    "requestReplyId": "f99ff93d-247e-4892-9d47-5895203b6532",
    "id": "0|com.android.mms|123|null|10040",
    "appName": "Messages",
    "isClearable": true,
    "ticker": "1410200505: FRM:sms@mc.internetmailserver.net,+ADw-sms+A\nMSG:",
    "title": "1410200505",
    "text": "FRM:sms@mc.internetmailserver.net,+ADw-sms+A\nMSG:",
    "time": "1541767165992"
  },
  "payloadSize": 8082,
  "payloadTransferInfo": {
  "port": 1750
}
```

I'm just about to go to bed, but I'll try and follow up with this later tomorrow. I'm not sure about the notification packet, but it's looking like we (GSConnect and KDE Connect) might have to offer a "legacy mode" for Samsung phones and possibly others. There are some weird rules in Android now, usually related to what app is the "Default SMS App" that might be preventing notifications as well as SMS history from being read.

**Comment by JohannPopper** [comment]
My apologies for my late reply. Very interesting indeed. You have figured out the discrepancy. Kubuntu 18.10 does not yet have version 1.3.2. It is still at 'kdeconnect/cosmic 1.3.1-1' in the respositories. Therefore, upon updating I would presumably lose legacy Samsung Android support with KDE Connect as well. 

As you expected, I had assumed sshfs was working on 18.10 GNOME with GSConnect as on Kubuntu 18.10, but it is not. Dependencies installed correctly, but console shows the following error, and Files > Internal Storage fails: 

WARNING: Samsung SAMSUNG-SM-J120A: sshfs: Unable to negotiate with 192.168.1.30 port 1739: no matching host key type found. Their offer: ssh-dss
JS ERROR: Gio.IOErrorEnum: Error when getting information for file “/run/user/1000/gsconnect/fd32cd259557efee/emulated/0”: Input/output error
openPath/<@/home/kiba/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/device.js:467:17
@/home/kiba/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/daemon.js:839:2

WARNING: Samsung SAMSUNG-SM-J120A: sshfs: read: Connection reset by peer

So that confirms the discrepancy was due to KDE Connect's new development. 

As for the state of the test phone when it showed the '"isCancel": true' event, it was on with the message app off. I double checked and tested again, and the same is reproduced with the phone on, all apps closed. There is no notification packet like the above example appearing in the terminal either before or after the Telephony packet. 

A legacy mode would be incredibly useful, albeit an unfortunate extra necessity. If possible I would also switch to whatever SMS GSConnect and KDE Connect would support, as messaging notification support is an outstanding feature.

**Comment by sredman** [comment]
I tend to think this is an Android-side problem, since the notification shown in https://github.com/andyholmes/gnome-shell-extension-gsconnect/issues/318#issuecomment-446137752 doesn't have a body. That's on the Android app to collect, and it is apparently unable to do so

> So that confirms the discrepancy was due to Kubuntu 18.10 being behind KDE Connect's new development.

It would be great if you could test on v1.3.3 of KDE Connect. Some bits of our handling for old SMS/Notification packets were changed, but overall much less has changed than in GSConnect
You can get v1.3.3 from the Kubuntu Backports repo: https://launchpad.net/%7Ekubuntu-ppa/+archive/ubuntu/backports/
Alternatively, if you have the resources, you could install a KDE Neon VM and test there. They have packaged KDE Connect v1.3.3

> A legacy mode would be incredibly useful, albeit an unfortunate extra necessity. If possible and convenient for me personally on the Android/service provider side, I would also switch to whatever messaging app GSConnect and KDE Connect would support, as messaging notification support is an outstanding feature of this project.
> 
> (The default SMS app on 2 of these phones is Samsung Messages; a third uses Samsung Messages+.)

Could you give Google Messages a shot? I don't know what part of our app-side changes would be giving us trouble with Samsung's Messages notifications, but I am using Google Messages successfully both with KDE Connect and GSConnect.

**Comment by JohannPopper** [comment]
I'm travelling on business, but will have more time to sit down and quickly respond over the weekend.

I did have time to test KDE Connect v1.3.3 on neon tonight. 

I was too hasty in my hope that we had found the source of the discrepancy between KDE Connect Android v1.3.1 pushing text notifications to Plasma, whereas GSConnect v13+ was not. Text notifications work perfectly on the latest version of neon, which I double-checked does indeed come with "kdeconnect/bionic,now 1.3.3-0xneon+18.04+bionic+build9 amd64 [installed,automatic]". 

I suppose this means it is not, at least directly, an Android-side problem, and there still must be some functional difference between the latest KDE Connect Plasma and the latest several versions of the GSConnect extension, that is somehow shutting down text notifications with at least some older Samsung phones using the Messages app, and that somehow results indicating the notification <"isCancel": true> on the remote device.

Unfortunately, I won't be able to try Google Messages until the weekend.

**Comment by JohannPopper** [comment]
Alright. Ran through some tests I hope may be useful for GSConnect. Three laptops -- one the latest neon Developer Stable Edition, one Kubuntu 18.10 w/Backports @5.14.3, one with Pop_OS! 18.10 with GSConnect. 

In summary: 

0) Updated my Samsung phones and KDE Connect Android app to the latest version; the phone used for the debug example below is now using Google Messages for SMS. *There was no change in behavior with Google Messages versus Samsung Messages with KDE Connect. However, with GSConnect, with Samsung Messages I receive no desktop SMS notifications at all; with Google Messages I receive SMS notifications, but they do not function as expected (described below in 2).*

1) Plasma KDE Connect on both neon and Kubuntu Backports receives texts correctly; that is, I get a notification upon receiving a text which I can click on that opens a text window showing the text with the Reply box, so that I may receive and send texts from my computer. 

2) GSConnect v17 on essentially Ubuntu 18.04 and 18.10, GNOME 3.28 & 3.30, respectively, using Google Messages on Android only, shows a notification in the GNOME notification area upon receiving a SMS. However, the expected behavior that is present in current KDE Connect fails -- as soon as I click on the notification, I get the blank "Type a message" window pictured in the screenshot here: 

https://i.imgur.com/b7J4r3H.png

Thus, a user would expect the text shown in notifications to be displayed in the "Type a message" window, as is the case with the latest KDE Connect text window that pops up after you click on the Plasma text notification. Without saving at least the most recently saved text, a user practically wouldn't be able read texts pushed to the desktop that are larger than the GNOME notification space, nor reference that text while typing a reply. 

So, there is still a discrepancy between KDE Connect and GSConnect, in that GSConnect only is not saving even single notifications for the Reply window. Since this is not a problem in Plasma, it would seem to be a separate, albeit related, bug from the deeper message history issue reported at #320 (which, incidentally, apparently occurs with Google Messages on a Samsung Phone as in the above screenshot, not only with Signal and Samsung Messages).

Below I've pasted excerpts from the GSConnect debug console using now Google Messages. The first segment is what you see immediately upon receiving a text which is then visible as a GNOME notification. 
 
```
}
DEBUG: [core.js:receive/<:404]: Samsung SAMSUNG-SM-J120A: {
  "id": 1545025654656,
  "type": "kdeconnect.notification",
  "body": {
    "payloadHash": "4aa98a49582cb77e89a87fa805f223e7",
    "id": "0|com.google.android.apps.messaging|0|com.google.android.apps.messaging:sms:47|10134",
    "appName": "Messages",
    "isClearable": true,
    "ticker": "(141) 010-0006: FRM:noreply@textem.net\nMSG:Thanks for your time, Andy and Simon!\n\n",
    "title": "(141) 010-0006",
    "text": "FRM:noreply@textem.net\nMSG:Thanks for your time, Andy and Simon!\n\n",
    "time": "1545025654626"
  },
  "payloadSize": 2474,
  "payloadTransferInfo": {
    "port": 1741
  }
}
DEBUG: [core.js:receive/<:404]: Samsung SAMSUNG-SM-J120A: {
  "id": 1545025655410,
  "type": "kdeconnect.telephony",
  "body": {
    "event": "sms",
    "messageBody": "FRM:noreply@textem.net\nMSG:Thanks for your time, Andy and Simon!\n\n",
    "phoneNumber": "1410100006"
  }
}
DEBUG: [core.js:receive/<:404]: Samsung SAMSUNG-SM-J120A: {
  "id": 1545025655459,
  "type": "kdeconnect.telephony",
  "body": {
    "event": "sms",
    "messageBody": "FRM:noreply@textem.net\nMSG:Thanks for your time, Andy and Simon!\n\n",
    "phoneNumber": "1410100006"
  }
}
```
As you can see below, as soon as I click on the GNOME text notification, a blank text Reply box appears (see screenshot above), and I get the *"isCancel":true*, presumably referring to the content that is not displayed therein.

```
DEBUG: [core.js:send:450]: Samsung SAMSUNG-SM-J120A: {
  "id": 1545025908048,
  "type": "kdeconnect.notification.request",
  "body": {
    "cancel": "0|com.google.android.apps.messaging|0|com.google.android.apps.messaging:sms:47|10134"
  }
}
DEBUG: [core.js:receive/<:404]: Samsung SAMSUNG-SM-J120A: {
  "id": 1545025910063,
  "type": "kdeconnect.notification",
  "body": {
    "id": "0|com.google.android.apps.messaging|0|com.google.android.apps.messaging:sms:47|10134",
    "isCancel": true
  }

```
So, maybe KDE Connect is doing an extra step to display the SMS in its reply box, whereas GSConnect is expecting a working message history to display texts in reply boxes. 

Please let me know if there's any other info you want. Thanks for you time.

**Comment by andyholmes** [comment]
It seems that the notification being sent lacks the `requestReplyId` field necessary for repliable notifications.

```js
DEBUG: [core.js:receive/<:404]: Samsung SAMSUNG-SM-J120A: {
  "id": 1545025654656,
  "type": "kdeconnect.notification",
  "body": {
    "payloadHash": "4aa98a49582cb77e89a87fa805f223e7",
    "id": "0|com.google.android.apps.messaging|0|com.google.android.apps.messaging:sms:47|10134",
    "appName": "Messages",
    "isClearable": true,
    "ticker": "(141) 010-0006: FRM:noreply@textem.net\nMSG:Thanks for your time, Andy and Simon!\n\n",
    "title": "(141) 010-0006",
    "text": "FRM:noreply@textem.net\nMSG:Thanks for your time, Andy and Simon!\n\n",
    "time": "1545025654626"
  },
  "payloadSize": 2474,
  "payloadTransferInfo": {
    "port": 1741
  }
}
```

> ...one the latest neon Developer Stable Edition, one Kubuntu 18.10 w/Backports @5.14.3...

Can you confirm that the version of KDE Connect on either of these is >= 1.3.2? There shouldn't be any discrepancy between the Android client sending `requestReplyId` to GSConnect or KDE Connect, so that leads me to believe that the version of KDE Connect is <= 1.3.1. Is it possible the old server is still running?

> ...whereas GSConnect is expecting a working message history to display texts in reply boxes.

GSConnect doesn't actually have a reply box; the window you're seeing is the full messaging app window. GSConnect uses Android-style "quick-reply" notifications when a notification carries the `requestReplyId`:

![screenshot from 2018-12-16 22-57-24](https://user-images.githubusercontent.com/1265208/50071361-5f710a80-0186-11e9-87a0-304d5ffacdfd.png)

Since notifications will only carry a "title" and "body", where title could be either a phone number or a name, it's nearly impossible to reliably respond to notifications if the `requestReplyId` field is missing.

If you're getting a truly repliable SMS dialog with KDE Connect, I would expect this is because it's a `kdeconnect.telephony` packet and not a `kdeconnect.notification` packet. @sredman can you confirm that `kdeconnect-kde <= 1.3.1` will show a notification that opens a dialog when it receives the legacy `kdeconnect.telephony` packet?

```js
DEBUG: [core.js:send:450]: Samsung SAMSUNG-SM-J120A: {
  "id": 1545025908048,
  "type": "kdeconnect.notification.request",
  "body": {
    "cancel": "0|com.google.android.apps.messaging|0|com.google.android.apps.messaging:sms:47|10134"
  }
}
DEBUG: [core.js:receive/<:404]: Samsung SAMSUNG-SM-J120A: {
  "id": 1545025910063,
  "type": "kdeconnect.notification",
  "body": {
    "id": "0|com.google.android.apps.messaging|0|com.google.android.apps.messaging:sms:47|10134",
    "isCancel": true
  }
```
These packets seem to be working as intended. The first is GSConnect advising Android that it has closed/acknowledged the notification, and the second is Android replying that it has closed that notification.

**Comment by JohannPopper** [comment]
Confirmed, repos and installed versions match: 
kdeconnect | 1.3.3-0ubuntu0.18.10.1~ubuntu18.10~ppa1 | Rik Mills (2018-11-17) for Kubuntu, and kdeconnect | Version: 1.3.3-0xneon+18.04+bionic+build9 for neon. 

"GSConnect uses Android-style "quick-reply" notifications when a notification carries the requestReplyId:" 

I see. Very nice. Assuming the Plasma distros are reporting their KDE Connect versions correctly, I guess the question is what else could be producing a repliable box in Plasma, since it would seem logical that a missing requestReplyId field would be an Android problem.

**Comment by sredman** [comment]
Upstream, we also have independent handling of SMS. This has been deprecated in favour of the handling by the Notifications plugin, since that can handle MMS as well as other messaging apps like WhatsApp, but it is still enabled by default, The plugin name in Android is, confusingly, "Send SMS". If you like, you could re-try the KDE Connect tests with that plugin disabled, but it's most likely that the Notifications plugin is not working there either because it's not like there are any special cases in Android to detect the desktop client.

**Comment by JohannPopper** [comment]
Ok, so I disabled the 'Send SMS' plugin Android-side, and KDE Connect 1.3.3. on both neon as well as Kubuntu 18.10 with Backports still worked as expected. I was able to read and reply on the desktop. Apparently, this means the issue cannot be due to GSConnect anticipating the deprecation of separate SMS handing. Yet, with notifications working in KDE Connect, and with text notifications now briefly visible with GSConnect if and only if I use Google Messages Android-side, I still do not receive repliable text notifications with GSConnect on any GNOME 3.30+ distro.

Here are a few screenshots of working repliable text notifications with KDE Connect 1.3.3 with the Android SMS plugin disabled (thus using only the now-standard notifications plugin):

1) https://i.imgur.com/4G4RdIu.png
2) https://i.imgur.com/vDqauS0.png

**Comment by sredman** [comment]
> Unless, of course, the bug is that the 'Send SMS' plugin cannot actually be deactivated on the Android app. It does seem to re-check itself upon unpairing and re-pairing the device.

Plugin settings are per-pairing, so that is intended behavior. Hopefully deactivating the plugin actually causes everything to be cleaned up as it should be, but that is a lightly-tested area!

I am about to ask for semi-redundant information, so sorry about that, but I am very confused by what is going on with this.

With Plasma/KDE Connect v1.3.3, (with both plugins enabled) could you please send an SMS to yourself and then screenshot the Plasmoid? It should look roughly like this:
![image](https://user-images.githubusercontent.com/1900703/50137952-18544980-025a-11e9-8374-8fd1bfdf3213.png)

**Comment by JohannPopper** [comment]
No problem. Anything I can do to help. 

Plasma/KDE Connect v1.3.3 plasmoid with both plugins enabled:

https://i.imgur.com/jcldK7G.png

**Comment by andyholmes** [comment]
> https://i.imgur.com/jcldK7G.png

For that screenshot it looks like Simon is right and the repliable notification is the older packet type. It's unfortunate that the notifications aren't being sent with a `requestReplyId`. I'll see if I can get a legacy fallback by next release, but it will certainly have less features.

**Comment by andyholmes** [comment]
I'm going to close this since I think it has been addressed with the new legacy UI in v18. Once installed you can enable legacy mode with the following command:

    dconf write /org/gnome/shell/extensions/gsconnect/device/<device-id>/plugin/sms/legacy true

You can get a list of device id's with the following command:

    dconf read /org/gnome/shell/extensions/gsconnect/devices

You should probably enabled/disable the SMS plugin after doing this. If this doesn't fix the problem, feel free to re-open the issue.

**Comment by JohannPopper** [comment]
I was away for the holiday. This is a wonderful surprise. Fantastic work. Thank you so much, Andy.

**Comment by JohannPopper** [comment]
Just to clarify my last comment, with the same testing conditions described above, I have been happily using v18-19 since release with Legacy mode enabled for these older Samsung phones, and this solution works very very well -- indeed more functional now than the default KDE Connect implementation, since GSConnect can initiate SMS from the desktop via the address book window. For my specific obsolete phone problem, at least, your new legacy mode seems to be the optimal solution. Thank you.

**Comment by andyholmes** [comment]
Heads up, when v20 is released the internal setting name has changed. So you will have to turn it back on, but there's a switch in the advanced settings page now so you don't have run anything in a terminal :)
